hazelcast-client/src/main/java/com/hazelcast/client/spi/ProxyManager.java

```diff
@@ -469,11 +467,10 @@ void set(Object o) {
         }
     }
 
-    private <T> T instantiateClientProxy(Class<T> proxyType, String serviceName, String id) {
+    private <T> T instantiateClientProxy(Class<T> proxyType, String serviceName, ClientContext context, String id) {
         try {
             final Constructor<T> constructor = proxyType.getConstructor(CONSTRUCTOR_ARGUMENT_TYPES);
-            return constructor.newInstance(serviceName, id);
-
+            return constructor.newInstance(serviceName, id, context);
```

Comment:
Here we add the `ClientContext` to the magical reflection call to create the proxies.